runner/execer/execers/sim.go

```diff
@@ -151,11 +159,19 @@ func (s *completeStep) run(status execer.ProcessStatus, p *simProcess) execer.Pr
 }
 
 type pauseStep struct {
-	wait *sync.WaitGroup
+	ch chan struct{}
 }
 
 func (s *pauseStep) run(status execer.ProcessStatus, p *simProcess) execer.ProcessStatus {
-	s.wait.Wait()
+	waitCh := make(chan struct{})
+	go func() {
+		p.Wait()
+		close(waitCh)
+	}()
+	select {
+	case <-waitCh:
```

Comment:
again comments:
case <-waitCh:  // is triggered by calling Abort(), which causes Wait() to close waitCh
case <-s.ch: // is triggered by calling Resume()